# Search evaluation

Run the current search-quality evaluation against the golden dataset. This uses the project's current corpus, embedding configuration, and default `k=5`; it does not compare configurations.

The Make target starts the database, loads the evaluation corpus into its isolated schema, and prints the current vector, text, and hybrid search metrics.

In [1]:
from pathlib import Path
from subprocess import run

project_root = Path.cwd().resolve()
if not (project_root / 'pyproject.toml').is_file():
    project_root = project_root.parents[1]

run(['make', 'evaluate-search'], cwd=project_root, check=True)

{
  "dataset": "evaluation/datasets",
  "corpus": "fixtures/corpus",
  "k": 5,
  "cases": 100,
  "modes": {
    "vector": {"hit_rate_at_k": 0.98, "recall_at_k": 0.98, "mean_reciprocal_rank": 0.9175, "mean_search_latency_ms": 34.33},
    "text": {"hit_rate_at_k": 0.25, "recall_at_k": 0.25, "mean_reciprocal_rank": 0.2383, "mean_search_latency_ms": 8.28},
    "hybrid": {"hit_rate_at_k": 0.98, "recall_at_k": 0.98, "mean_reciprocal_rank": 0.9275, "mean_search_latency_ms": 40.99}
  }
}


**Conclusion:** The current hybrid search is the strongest retrieval choice: it matches vector search's 98% hit rate and recall while slightly improving MRR (0.9275 versus 0.9175), for a modest 6.7 ms mean-latency increase. Full-text search is much faster but retrieves relevant evidence for only 25% of cases, so it is not suitable as the current standalone default.